File: src/main/java/com/mystchonky/arsoscura/common/init/ArsOscuraLang.java
package com.mystchonky.arsoscura.common.init;

import com.mystchonky.arsoscura.ArsOscura;
import com.tterrag.registrate.Registrate;
import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.MutableComponent;

public class ArsOscuraLang {

    private static final Registrate REGISTRATE = ArsOscura.registrate();
    public static final Component SIGIL_EMPTY = REGISTRATE.addLang("tooltip", ArsOscura.prefix("sigil_empty"), "No entity stored");
    public static final MutableComponent SIGIL_WITH_ENTITY = REGISTRATE.addLang("tooltip", ArsOscura.prefix("sigil_with_entity"), "Entity stored: %s");



    public static void register() {
        Integrations.registeredSpells.forEach(spell -> {
            REGISTRATE.addRawLang(ArsOscura.MODID + ".glyph_name." + spell.getRegistryName().getPath(), spell.getName());
            REGISTRATE.addRawLang(ArsOscura.MODID + ".glyph_desc." + spell.getRegistryName().getPath(), spell.getBookDescription());
        });
    }
}

File: src/main/java/com/mystchonky/arsoscura/common/init/ArsOscuraLang.kt
package com.mystchonky.arsoscura.common.init

import com.hollingsworth.arsnouveau.api.spell.AbstractSpellPart
import com.mystchonky.arsoscura.ArsOscura
import com.mystchonky.arsoscura.ArsOscura.registrate
import com.mystchonky.arsoscura.integration.bloodmagic.BloodMagicIntegration
import net.minecraft.network.chat.Component
import net.minecraft.network.chat.MutableComponent
import net.minecraft.resources.ResourceLocation
import java.util.function.Consumer

object ArsOscuraLang {
    private val REGISTRATE = registrate()
    val prefix: (String) -> ResourceLocation = { ArsOscura.prefix(it) }
    val SIGIL_EMPTY: Component = REGISTRATE.addLang("tooltip", prefix("sigil_empty"), "No entity stored")
    val SIGIL_WITH_ENTITY: MutableComponent =
        REGISTRATE.addLang("tooltip", prefix("sigil_with_entity"), "Entity stored: %s")

    // Blood Magic Integration
    val SCHOOL_BLOODMAGIC: Component =
        REGISTRATE.addRawLang("ars_nouveau.school." + BloodMagicIntegration.BLOODMAGIC.id, "Blood Magic")
    val LOW_LP: Component = REGISTRATE.addLang("alert", prefix("no_lp"), "Your soul feels weak..")
    val SERENE_EFFECT: Component = REGISTRATE.addLang("effect", prefix("serene"), "Serene")
    val MANA_BONUS_UPGRADE: Component = REGISTRATE.addLang("living_upgrade", prefix("mana_bonus"), "Mana Attunement")


    fun register() {
        ArsNouveauIntegration.registeredSpells.forEach(Consumer { spell: AbstractSpellPart ->
            REGISTRATE.addRawLang(ArsOscura.MODID + ".glyph_name." + spell.registryName.path, spell.getName())
            REGISTRATE.addRawLang(ArsOscura.MODID + ".glyph_desc." + spell.registryName.path, spell.bookDescription)
        })
    }
}
